angular/src/directives/control-value-accessors/text-value-accessor.ts

```diff
@@ -21,6 +21,27 @@ export class TextValueAccessorDirective extends ValueAccessor {
 
   @HostListener('ionChange', ['$event.target'])
   _handleInputEvent(el: any): void {
-    this.handleChangeEvent(el, el.value);
+    this.commitValueChange(el, el.value);
+  }
+}
+
+@Directive({
+  selector: 'ion-input:not([type=number])',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: InputValueAccessorDirective,
+      multi: true,
+    },
+  ],
+})
+export class InputValueAccessorDirective extends ValueAccessor {
```

Comment:
Why do we exclude `type="number"`?

Reply:
`ion-input[type="number"]` has a separate directive [implementation](https://github.com/ionic-team/ionic-framework/blob/4c13e81679ea79f3a99f7268f6fa0e35251baf43/angular/src/directives/control-value-accessors/numeric-value-accessor.ts#L7). The numeric value accessor parses the value to a float/numeric value.

Reply:
Oh that's right. Thanks!